app/models/hardware.rb

```diff
@@ -51,15 +51,19 @@ def mac_addresses
     @mac_addresses ||= nics.collect(&:address).compact.uniq
   end
 
-  def ram_size_in_bytes
-    memory_mb.to_i * 1.megabyte
-  end
-  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
-  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
-    t.grouping(Arel::Nodes::Multiplication.new(Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]),
-                                               1.megabyte))
+  # resulting sql: "(("hardwares"."ram_size_in_bytes") / 1048576)"
+  virtual_attribute :memory_mb, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Division.new([t[:ram_size_in_bytes]], 1024 * 1024))
   end)
 
+  def memory_mb=(memory_mb)
+    @memory_mb = memory_mb
+  end
+
+  def memory_mb
+    @memory_mb
```

Comment:
maybe make 1024*1024 a constant?

```suggestion
  ram_size_in_bytes / (1024 * 1024)
```